packages/plugin-core/src/commands/InsertNoteIndexCommand.ts

```diff
@@ -0,0 +1,78 @@
+import { DENDRON_COMMANDS } from "../constants";
+import { BasicCommand } from "./base";
+import { VSCodeUtils } from "../utils";
+import { DNodeUtils, NoteProps } from "@dendronhq/common-all";
+import { getEngine } from "../workspace";
+import { window } from "vscode";
+import _ from "lodash";
+
+type CommandOpts = {
+  /*
+   * optional flag that will wrap the index block with markers.
+   */
+  marker?: boolean;
+};
+
+type CommandOutput = CommandOpts;
+
+export class InsertNoteIndexCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  key = DENDRON_COMMANDS.INSERT_NOTE_INDEX.key;
+
+  // TODO: make this into a util once the cli version is implemented.
+  // NOTE: the marker flag is not exposed to the plugin yet.
+  genNoteIndex(notes: NoteProps[], opts: {
+    marker?: boolean
+  }) {
+    const listItems = notes.map((note) => `- [[${note.title}\|${note.fname}]]`)
```

Comment:
for the index, what about the case with the same note in multiple vaults?
also, should [noXvaultWikiLink](https://wiki.dendron.so/notes/f83c1d87-eac0-48f3-a5cf-8a69989d8ec1.html#noxvaultwikilink) impact how links are generated?

Reply:
You're right. I changed it so that it uses `createWikiLink` so that it can add the vault prefix as necessary.